File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/DivergentTasks.java
package com.xebia.functional.xef.java.auto;

import java.util.concurrent.ExecutionException;

public class DivergentTasks {

    public Long numberOfMedicalNeedlesInWorld;

    public static void main(String[] args) throws ExecutionException, InterruptedException {
        try (AIScope scope = new AIScope()) {
            scope.prompt("Provide the number of medical needles in the world", DivergentTasks.class)
                  .thenAccept(numberOfNeedles -> System.out.println("Needles in world:\n" + numberOfNeedles.numberOfMedicalNeedlesInWorld))
                  .get();
        }
    }

}

File: src/jvmTest/kotlin/com/xebia/functional/examples/auto/DivergentTasks.kt
package com.xebia.functional.examples.auto

import com.xebia.functional.auto.agents.WikipediaAgent
import com.xebia.functional.auto.ai
import kotlinx.serialization.Serializable

@Serializable
data class NumberOfMedicalNeedlesInWorld(val numberOfNeedles: Long)

suspend fun main() {
    val needlesInWorld: NumberOfMedicalNeedlesInWorld = ai(
        """|Provide the number of medical needles in the world.
        """.trimMargin(),
        agents = listOf(WikipediaAgent)
    )
    println("Needles in world: ${needlesInWorld.numberOfNeedles}")
}

